packages/lexical/src/LexicalNode.ts

```diff
@@ -198,6 +232,19 @@ export class LexicalNode {
   __prev: null | NodeKey;
   /** @internal */
   __next: null | NodeKey;
+  /** @internal */
+  __state: DeepImmutable<State> = {};
+
+  getState<T extends StateKey>(k: T): T['value'] {
+    const self = this.getLatest();
+    // If the state is not set, return the default value
+    return self.__state[k.key] ?? k.parse(undefined);
+  }
+
+  setState<T extends StateKey>(k: T, v: T['value']) {
+    const self = this.getWritable();
+    self.__state = {...self.__state, [k.key]: v};
+  }
```

Comment:
should return `this` for chaining like other mutator methods

```suggestion
  setState<T extends StateKey>(k: T, v: T['value']): this {
    const self = this.getWritable();
    self.__state = {...self.__state, [k.key]: v};
    return self;
  }
```